### ISSUE TITLE
Bump RenderingServer.MAX_MESH_SURFACES

### ISSUE THREAD

**Issue by Goutte** [godotengine/godot-proposals#5656]
### Describe the project you are working on

I'm working on a modern Go Game playable on more "natural" terrains.

Here's a screenshot of the Godot 3.x demo with a boring wood texture and ugly stones, but already an interesting terrain, gameplay-wise :

![Screenshot_20221001_194109](https://user-images.githubusercontent.com/592545/197929348-2724b11c-0cee-4b21-aba2-92d5e6948375.png)

> By the way, if there are Go players here whose interest is piqued, ping me for an early access !

### Describe the problem or limitation you are having in your project

I need to upgrade to Godot 4 to enable online games (targeting HTML5 export as well), to benefit from `Callable`s in my REST OAS generated client, and a headless referee.

I'm hitting the `MAX_MESH_SURFACES` at `256` on the geode, which tries to generate about `600`.  It's **quite convenient** for me to have **each tile as a surface**, all in one mesh.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Is there a strong case against raising this `MAX_MESH_SURFACES` default limit above `256` ?  (to, say, `1024` ?)
What's the cost, here ?

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

`MAX_MESH_SURFACES = 1024` [here](https://github.com/godotengine/godot/blob/bff9fcfc1c90f5e121a9f9ae0af93616b5c7304d/servers/rendering_server.h#L85)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Would need compilation of custom Godot.

### Is there a reason why this should be core and not an add-on in the asset library?

It's a constant.

**Comment by clayjohn** [comment]
I don't know of a reason to have the 256 surface limit other than to avoid pathological cases where you accidentally continue adding surfaces forever. 

I guess it can be made configurable like we did with max lights per object in 3.x: https://github.com/godotengine/godot/pull/43606

**Comment by Goutte** [comment]
That's great news !   Thanks @clayjohn for the link to the related PR, this is what I needed to try and take this on myself.

And thanks @Calinou for the title edit, I'll keep it in mind next time.

**Comment by Goutte** [comment]
This appears to be a little more intricate than increasing the value.

Here's what I get with a value of `256` :

![Screenshot_20221103_164232](https://user-images.githubusercontent.com/592545/199768491-9d7f14c6-9c3d-4d5a-ab0c-fdd0cce622a5.png)

> And plenty of errors due to trying to add more than 256 surfaces

I've managed to add it to project settings (under `rendering/limits/geometry/max_mesh_surfaces`), and read from it wherever `MAX_MESH_SURFACES` was used before, and as a result of setting the value to `1024` for example I get this :

![Screenshot_20221103_164326](https://user-images.githubusercontent.com/592545/199768558-246675c5-f11b-41ac-b9fd-d43f1e1e21a2.png)

> And no more errors.

Either I [wrote this wrong](https://github.com/Goutte/godot/tree/configure-max-mesh-surfaces), or there's another limit somewhere.

**Comment by Goutte** [comment]
I tried looking at arrays (`mesh->surfaces` does have `600` elements), and for `uint8_t` in case that's where the other part of the limit was, but I can't find why only 256 surfaces are rendered (and pseudo-random ones at that).

It might be something is wrong with my demo (but I double checked), or I've overlooked an `array` or a `uint8` (likely), or perhaps it's deeper (pseudo-random suggests it is ?).

This is not a blocking issue for me (I've worked around it), and I'm running out of steam for this, so I will let this rest for now.

> :bulb: If you know why this is happening, please do enlighten us !

**Comment by wnhrt** [comment]
> I don't know of a reason to have the 256 surface limit other than to avoid pathological cases where you accidentally continue adding surfaces forever. 
> 
> I guess it can be made configurable like we did with max lights per object in 3.x: https://github.com/godotengine/godot/pull/43606

@clayjohn could this fit in 4.0?

**Comment by clayjohn** [comment]
@wnhrt No, as we are currently in a [feature freeze](https://godotengine.org/en/article/godot-4-0-development-enters-feature-freeze) the focus is on fixing bugs before releasing 4.0 rather than implementing proposals
